### PR TITLE
Honor child widget resizes in UIViewBox

### PR DESCRIPTION
**Pull request by swankjesse**
There’s still some real problems in `UIViewBox`, but now we’re appropriately responding when a child view’s intrinsic dimensions changes.

### CODE REVIEW

File: redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt [hunk 1]
```diff
@@ -67,45 +72,62 @@ internal class UIViewBox : Box<UIView> {
...
+        setNeedsLayout() // Update layout of subviews.
+      }
+    }
+
     override fun layoutSubviews() {
       super.layoutSubviews()
 
       children.widgets.forEach { widget ->
         val view = widget.value
+        view.setFrame(frame)
```
Comment (swankjesse) [inline redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt:130]: This is a critical fix. Without this line the preceding size sticks.

File: redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiBoxTest_testDynamicWidgetResizing[LTR]_v1.png
Comment (JakeWharton) [inline redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiBoxTest_testDynamicWidgetResizing[LTR]_v1.png:1]: This one is vertical while UIKit and Views are horizontal. Existing bug, I assume.

File: redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt [hunk 2]
```diff
@@ -41,21 +42,25 @@ import platform.UIKit.UIEdgeInsetsMake
...
+internal class UIViewBox :
+  Box<UIView>,
+  ResizableWidget<UIView> {
   override val value: View = View()
 
   override var modifier: Modifier = Modifier
 
   override val children get() = value.children
 
+  override var sizeListener: ResizableWidget.SizeListener? by value::sizeListener
```
Comment (JakeWharton) [inline redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt:54]: This used to produce wildly inefficient garbage code, but at some point it no longer does. Nice!

https://kotlin.godbolt.org/z/5heMY7KMr

File: redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt (hunk 1 above)
Comment (JakeWharton) [reply redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt:130]: I assume this is where we need to apply margins and adjust each child's frame to fix #2003

**Review by JakeWharton — APPROVED** [review]

**Pull request merged**